internal/controller/genstatus.go

```diff
@@ -270,9 +292,12 @@ func (r *TemporalWorkerDeploymentReconciler) generateStatus(ctx context.Context,
 	defaultVersionID = routingConfig.CurrentVersion
 
 	// Check if the worker deployment was modified out of band of the controller (eg. via the Temporal CLI)
-	if workerDeploymentInfo.LastModifierIdentity != "temporal-worker-controller" &&
-		workerDeploymentInfo.LastModifierIdentity != "" {
-		// TODO(jlegrone): if it was set by another client, switch to manual mode
+	if wasModifiedExternally(&workerDeploymentInfo) {
```

Comment:
I think there's a race condition if configuration on the Temporal side is updated _after_ the controller calls the API to get worker deployment info. In that situation, this condition wouldn't evaluate `true` and we would proceed to potentially overwrite the out of band changes.

In a previous Temporal API version there was a `ConflictToken` field that could be used to avoid this: https://github.com/DataDog/temporal-worker-controller/blob/6d4cc8642ee14301de233ad8f8d506c874ace74f/internal/controller/execplan.go#L65

Is there a way to address the issue with current APIs?